sdk/servicebus/service-bus/test/renewLock.spec.ts

```diff
@@ -68,24 +60,53 @@ describe("Message Lock Renewal", () => {
     }
   );
 
-  it(testClientType + ": cannot renew lock on peeked message", async function(): Promise<void> {
-    const receiver = await serviceBusClient.test.createPeekLockReceiver(autoGeneratedEntity);
+  for (let index = 0; index < 1000; index++) {
+    testClientType = getRandomTestClientTypeWithNoSessions();
 
-    const testMessage = TestMessage.getSample();
-    await sender.sendMessages(testMessage);
+    beforeEach(async () => {
```

Comment:
Won't this cause 1000 beforeEach functions to be registered since you're running this inside a for loop? I suspect this is part of why you're running out of memory.

Reply:
Good point on the memory. Though it happened only on one of the entries of the matrix, it did happen.
 
And this won't make it to the final fix as there won't be a for-loop.

How does the fix in the `it()` looks like?

Reply:
I'm not sure what the original issue was (no description in PR) but if it was that you weren't getting a message to renew the lock on, this seems fine. I thought there was already a 1 minute wait in peekMessages though?

Reply:
But the service is returning 0 messages and thus the wait time doesn't matter.

Reply:
Here's the fix to review - https://github.com/Azure/azure-sdk-for-js/pull/13346